pkg/operator/controller/instance_controller.go

```diff
@@ -822,6 +857,25 @@ func (r *InstanceReconciler) updateCondition(conditions map[string]metav1.Condit
 	return update
 }
 
+// triggerAnotherInstance checks if another CRD instance exists and triggers it.
+func (r *InstanceReconciler) triggerAnotherInstance(ctx context.Context, name, namespace string) error {
+	List := &clusterlink.InstanceList{}
+	if err := r.List(ctx, List); err != nil {
```

Comment:
Why not call `List` on the current namespace only? Is there a way to set the List to do that?

Reply:
It doesn't matter. All the items are in the operator namespace.

Reply:
True for now. But from k8s API point of view you're giving it a constraint to work with an optimize the listing so it does not need to consider objects in other namespaces.
In addition, constraining to the current workspace would work correctly even with per-namespace deployment (only look for replacement in the same namespace as the one deleted - RBAC in this mode would be extra protection since you wont have a way to list outside the current namespace, but I'm not sure if you'll get an access error or just the instances in namespaces you're allowed to access)